### ISSUE TITLE
val/seg_loss nan

### ISSUE THREAD

**Issue by Kader95Sh** [ultralytics/ultralytics#1615]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

 val/box_loss,           val/seg_loss,           val/cls_loss,           val/dfl_loss,
1.5397,                    nan,                 1.0328,                  1.438,
1.5804,                    nan,                 1.0301,                 1.4785,
1.5696,                    nan,                 1.0647,                 1.4974,
1.5637,                    nan,                 1.0311,                 1.4929,

### Environment

Ultralytics YOLOv8.0.56  Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (Tesla V100-SXM2-32GB, 32501MiB)
                                                            CUDA:1 (Tesla V100-SXM2-32GB, 32501MiB)
                                                            CUDA:2 (Tesla V100-SXM2-32GB, 32501MiB)
                                                            CUDA:3 (Tesla V100-SXM2-32GB, 32501MiB)
[34m[1myolo/engine/trainer: [0mtask=segment, mode=train, model=yolov8x-seg.pt, data=./custom2.yaml, epochs=800, patience=50, batch=16, imgsz=1408, save=True, save_period=50, cache=False, device=0,1,2,3, workers=16, project=kitti360V2, name=SegX1408, exist_ok=False, pretrained=False, optimizer=SGD, verbose=True, seed=0, deterministic=True, single_cls=False, image_weights=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, hide_labels=False, hide_conf=False, vid_stride=1, line_thickness=3, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, fl_gamma=0.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, v5loader=False, tracker=botsort.yaml, save_dir=kitti360V2/SegX1408
Overriding model.yaml nc=80 with nc=18

### Minimal Reproducible Example

yolo task=segment mode=train model=yolov8x-seg.pt device=\'0,1,2,3\' data=./custom2.yaml epochs=800 batch=16 imgsz=1408 save_period=50 workers=16 project=kitti360V2 name=SegX1408 patience=50

### Additional

Hallo,

I am having problem with val/seg_loss. It is always nan.
The labels are as mask for every Object.    .....  . So i am just taking every Pixel for the object and putting it in labels.

In the Train/seg_loss every this looks good 

train/seg_loss,
3.1815,
2.8455,
2.8549,
2.8117,
2.6921,

I did so far 3 Tries and all having same Problem.

Thank you for your answers.

Note: my train and val classes having imbalance numbers. Could be that the problem?

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Laughing-q** [comment]
@Kader95Sh I'm able to reproduce the issue. I opened a PR [https://github.com/ultralytics/ultralytics/pull/1618](https://github.com/ultralytics/ultralytics/pull/1618) and it works to me.
could you try the training again with the PR? Thanks!

**Comment by Kader95Sh** [comment]
Thank you for the fast answer. I tried again after PR. the problem still here
val/seg_loss,    train/seg_loss,
nan                   3.1815,

**Comment by glenn-jocher** [comment]
@Kader95Sh we'll merge #1618 anyway as it might provide some improvement. nan val losses won't directly affect your training, so you probably have nothing to worry about in reality. 

If you could help us debug the source of the nan results though please let us know what we could update in the loss function to resolve this.

Alternatively please check your val labels to ensure there are no issues there.

**Comment by Laughing-q** [comment]
@Laughing-q I was able to reproduce the issue before the fix with the following command:
```python
yolo segment train model=yolov8n-seg.pt epochs=2 data=runs/coco128.yaml imgsz=640 batch=4
```
the nan val_loss shows because there are backgrounds in my test datasets then the val loss of these background becomes nan while validating.
But in my case the nan val_loss is gone after the fix PR.
